pkg/rabbitmq/impl/rabbitmqcluster.go

```diff
@@ -0,0 +1,119 @@
+package impl
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	rabbitmqv2 "github.com/rabbitmq/cluster-operator/v2/api/v1beta1"
+
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// NewRabbitMqCluster returns an initialized RabbitmqCluster.
+func NewRabbitMqCluster(
+	rabbit *rabbitmqv2.RabbitmqCluster,
```

Comment:
@olliewalsh should we pass here the `serviceLabels` we generate from the controller and that are used as Topology defaultLabelSelector? I had this comment in a different (probably the wrong) place, but I think here is more appropriate.
I imagine the flow as:
1. we generate `serviceLabels` in rabbitmq_controller.go
2. we call `NewRabbitMqCluster` and we pass down those `serviceLlabels`
3. we assign them to `.Spec.Selector`.

what do you think about this? Does it make sense?